ember-yeti-table/src/components/yeti-table/thead/row.gjs

```diff
@@ -40,27 +40,14 @@ export default class THeadRow extends Component {
   cells = [];
 
   registerCell(cell) {
-    let column;
-
-    if (cell.prop) {
-      column = this.args.columns.findBy('prop', cell.prop);
-      cell.column = column;
-    } else {
-      let index = this.cells.length;
-      column = this.args.columns[index];
-
-      return column;
-    }
```

Comment:
Seems like we're removing this logic of finding the column by prop and setting it into the cell.
I **think** that only using the index won't work, because you can render columns conditionally, which means, "out of order".

If you do something like
```hbs
<YetiTable @data={{this.data}} as |t|>

  <t.header as |h|>
    <h.column @prop="firstName">
      First name
    </h.column>
  
    {{#if this.showPoints}}
      <h.column @prop="points">
        Points
      </h.column>
    {{/if}}
    
    <h.column @prop="lastName">
      Last name
    </h.column>
  </t.header>

  <t.body/>

</YetiTable>
```

would the new cells have access to the correct column?

I know you're probably supposed to use `<h.column @prop="points" @visible={{this.showPoints}}>` for this case, but it doesn't change the fact that it's still a breaking change. And I guess this problem gets more aggravated if people render columns in loops, e.g

```hbs
<YetiTable @data={{this.data}} as |t|>

  <t.header as |h|>
    <h.column @prop="firstName">
      Fixed Column
    </h.column>
    
    {{#each this.columns as |c|}}
      <h.column @prop={{c.prop}}>
       {{c.label}}
      </h.column>
    {{/each}}
    
    <h.column @prop="lastName">
      Fixed Column
    </h.column>
  </t.header>

  <t.body/>

</YetiTable>
```

Reply:
This update was for cell which you dont actually use in the example. If you were to do another header row with cell, then the IF in your first example would have to be reproduced in the second header row that uses cell and you would render 2 TH(s) or 3 TH(s) in each row. In both cases the indexes still align. Same with the loop. 

If you render a different number of TH(s) in each row, that will be a problem anyway unless you use colspan. I pointed out that colspan does not work at all. It would require more code changes (and yes I pointed out that prop would have to be used and mandatory for colspan to work). 

This PR was not attempting to fix colspan but only the issue that the second header does not work when there is a second header using cell. So not really a breaking change, since it does not work now anyway. 

As for the IF or using visible, I think that they are effectively the same thing given that the visible does the if https://github.com/miguelcobain/ember-yeti-table/blob/master/ember-yeti-table/src/components/yeti-table/thead/row/column.gjs#L46. Either way there is no DOM inserted. The difference is if you do the if, you have to do it in the header and any secondary headers, and if you re using the block version of body and footer, where as the visible allows you to specify it at the header only and not have to reproduce the if in second headers and the block version of body and footer

Reply:
I fixed the regression for mapping additional header cells to columns via `@prop` as well in 779925703b861f332d34ba6454597b402492542e.